lib/utils/lru-cache/lru_cache.go

```diff
@@ -93,3 +119,27 @@ func (c *LRUCache[K, V]) Put(key K, value V) {
 	newElem := c.lruList.PushFront(newEntry)
 	c.cache[key] = newElem
 }
+
+// Put adds a key-value pair to the cache.
+func (c *LRUCache[K, V]) Delete(key K) bool {
+	c.Lock()
+	defer c.Unlock()
+
+	val, exists := c.cache[key]
+	if !exists {
+		return false
+	}
+
+	c.lruList.Remove(val)
+
+	delete(c.cache, key)
+	return true
+}
+
+// Put adds a key-value pair to the cache.
+func (c *LRUCache[K, V]) Len() int {
+	c.Lock()
+	defer c.Unlock()
+
+	return len(c.cache)
```

Comment:
lruList has Len() attribute, it is 100% O(1) complexity while I am not sure about bult-in function

Reply:
Yes, it is O(1), [go keeps track of the items amount internally](https://github.com/golang/go/blob/master/src/runtime/map.go#L120)